flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/DataStreamProcTimeAggregateGlobalWindowFunction.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.aggregate
+
+import org.apache.flink.streaming.api.functions.windowing.RichAllWindowFunction
+import org.apache.flink.types.Row
+import org.apache.flink.util.Collector
+import org.apache.flink.streaming.api.windowing.windows.Window
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.table.functions.Accumulator
+
+import java.lang.Iterable
+import org.apache.flink.table.functions.AggregateFunction
+
+/**
+  * Computes the final aggregate value from incrementally computed aggreagtes.
+  *
+  * @param aggregates The aggregates to be computed
+  * @param aggFields the fields on which to apply the aggregate.
+  * @param forwardedFieldCount The fields to be carried from current row.
+  */
+class DataStreamProcTimeAggregateGlobalWindowFunction[W <: Window](
```

Comment:
@fhueske 

I will start with the second part - supporting LIMIT, Distinct, top.... even if these leads to a separate operator that would be applied in theory before we should optimize and combine them...it does not make sense to have 2 operators for one logical operation. Secondly if we refer to keeping them separately - the logic would be that you have a window where you keep all the data to apply limit/top/... and than you fire multiple elements that you selected to the operator to compute the aggregator - so you would reset the aggregator every time.

Regarding your remark of managing the state in process function by ourselves...we are basically creating a window if we keep all elements in the state of the process function. we still need to go through the potentially all elements to evict them and update the aggregator. I am not sure which is more efficiently managed: a large function state or the buffered elements of the window. I would assume in the best case they are similar in performance... if window buffers are not optimized. Nevertheless...there is no advantage as far as i see and understand the problem of having the process function ...but i am sure it brings extra overhead and limitations in the future.